lib/android-tools-info.ts

```diff
@@ -3,25 +3,63 @@
 
 import * as path from "path";
 import * as semver from "semver";
+import {EOL} from "os";
 
 export class AndroidToolsInfo implements IAndroidToolsInfo {
 	private static ANDROID_TARGET_PREFIX = "android";
 	private static SUPPORTED_TARGETS = ["android-17", "android-18", "android-19", "android-21", "android-22", "android-23"];
 	private static MIN_REQUIRED_COMPILE_TARGET = 22;
 	private static REQUIRED_BUILD_TOOLS_RANGE_PREFIX = ">=22";
-	private static VERSION_REGEX = /^(\d+\.){2}\d+$/;
+	private static VERSION_REGEX = /((\d+\.){2}\d+)/;
+	private static MIN_JAVA_VERSION = "1.7.0";
+	private static ANDROID_EXECUTABLE_NAME = "android";
+
 	private showWarningsAsErrors: boolean;
 	private toolsInfo: IAndroidToolsInfoData;
 	private selectedCompileSdk: number;
 	private installedTargetsCache: string[] = null;
 	private androidHome = process.env["ANDROID_HOME"];
+	private pathToAndroidExecutable: string;
 
 	constructor(private $childProcess: IChildProcess,
 		private $errors: IErrors,
 		private $fs: IFileSystem,
+		private $hostInfo: IHostInfo,
 		private $logger: ILogger,
 		private $options: IOptions) {}
 
+	public getPathToAndroidExecutable(): IFuture<string> {
+		return ((): string => {
+			if (!this.pathToAndroidExecutable) {
+				if(!this.validateAndroidHomeEnvVariable(this.androidHome).wait()) {
+					let androidPath = path.join(this.androidHome, "tools", AndroidToolsInfo.ANDROID_EXECUTABLE_NAME) + (this.$hostInfo.isWindows ? ".bat" : "");
+					if(!this.trySpawnAndroidHelp(androidPath).wait() && !this.trySpawnAndroidHelp(AndroidToolsInfo.ANDROID_EXECUTABLE_NAME).wait()) {
+						this.$errors.failWithoutHelp(`Unable to find "${AndroidToolsInfo.ANDROID_EXECUTABLE_NAME}" executable file. Make sure you have set ANDROID_HOME environment variable correctly.`);
+					}
+				} else {
+					this.$errors.failWithoutHelp("ANDROID_HOME environment variable is not set.");
+				}
+			}
+
+			return this.pathToAndroidExecutable;
+		}).future<string>()();
+	}
+
+	private trySpawnAndroidHelp(androidPath: string): IFuture<boolean>{
```

Comment:
Isn't it better to name this method `trySetAndroidPath`? It is unclear when reading `getPathToAndroidExecutable` that `trySpawnAndroidHelp` actually sets the path to the executable.